Compiling error dued to openPMDWriter interface

Trying to compile the new `dev` code base with the CUDA 13 compiler i got this error linked to the `openPMDWriter` 

````
/lustre/rz/dbertini/hydra/pog/picongpu/include/picongpu/plugins/openPMD/openPMDWriter.x.cpp:1860:17:   required from here
/lustre/rz/dbertini/hydra/pog/picongpu/include/picongpu/plugins/openPMD/openPMDWriter.x.cpp:870:140: error: dependent-name ‘pmacc::traits::GetComponentsType<pmacc::math::Vector<float, 1, pmacc::math::ArrayStorage<float, 1> >, is_fundamental_v<pmacc::math::Vector<float, 1, pmacc::math::ArrayStorage<float, 1> > > >::type’ is parsed as a non-type, but instantiation yields a type
  870 |                     openPMDWriter::template writeField<ComponentType>(
      |                                                                                                                                            ^   
/lustre/rz/dbertini/hydra/pog/picongpu/include/picongpu/plugins/openPMD/openPMDWriter.x.cpp:870:140: note: say ‘typename pmacc::traits::GetComponentsType<pmacc::math::Vector<float, 1, pmacc::math::ArrayStorage<float, 1> >, is_fundamental_v<pmacc::math::Vector<float, 1, pmacc::math::ArrayStorage<float, 1> > > >::type’ if a type is meant
/lustre/rz/dbertini/hydra/pog/picongpu/include/picongpu/plugins/openPMD/openPMDWriter.x.cpp:870:146: error: no matching function for call to ‘picongpu::openPMD::openPMDWriter::<expression error>(picongpu::openPMD::ThreadParams*&, const uint32_t&, const uint32_t&, std::string, std::__shared_ptr_access<picongpu::FieldTmp, __gnu_cxx::_S_atomic, false, false>::element_type&, std::vector<double>, std::vector<double>, std::remove_reference<std::vector<std::vector<float> >&>::type, const float_X&, const bool&, bool)’
  870 |                     openPMDWriter::template writeField<ComponentType>(
      |                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                                                           ^                                                                                                                                                                                     
/lustre/rz/dbertini/hydra/pog/picongpu/include/picongpu/plugins/openPMD/openPMDWriter.x.cpp:1532:1: note: candidate: ‘template<class ComponentType, class FieldBuffer> static void picongpu::openPMD::openPMDWriter::writeField(picongpu::openPMD::ThreadParams*, uint32_t, uint32_t, std::string, FieldBuffer&, std::vector<double>, std::vector<double>, std::vector<std::vector<float> >, picongpu::float_X, bool, bool)’
 1532 |             static void writeField(
      | ^         
/lustre/rz/dbertini/hydra/pog/picongpu/include/picongpu/plugins/openPMD/openPMDWriter.x.cpp:1532:1: note:   template argument deduction/substitution failed:
/lustre/rz/dbertini/hydra/pog/picongpu/include/picongpu/plugins/openPMD/openPMDWriter.x.cpp: In instantiation of ‘void picongpu::openPMD::openPMDWriter::GetFields<picongpu::FieldTmpOperation<Solver, Species, Filter> >::operator_impl(picongpu::openPMD::ThreadParams*, uint32_t) [with Solver = picongpu::particles::particleToGrid::ComputeGridValuePerFrame<picongpu::particles::shapes::TSC, picongpu::particles::particleToGrid::derivedAttributes::EnergyDensity>; Species = picongpu::Particles<pmacc::meta::String<'e'>, boost::mp11::mp_list<picongpu::particlePusher<picongpu::particles::pusher::Boris, pmacc::pmacc_isAlias>, picongpu::shape<picongpu::particles::shapes::TSC, pmacc::pmacc_isAlias>, picongpu::interpolation<picongpu::FieldToParticleInterpolation<picongpu::particles::shapes::TSC, picongpu::AssignedTrilinearInterpolation>, pmacc::pmacc_isAlias>, picongpu::current<picongpu::currentSolver::Esirkepov<picongpu::particles::shapes::TSC, picongpu::currentSolver::strategy::CachedSupercells, 3>, pmacc::pmacc_isAlias>, picongpu::massRatio<picongpu::MassRatioElectrons, pmacc::pmacc_isAlias>, picongpu::chargeRatio<picongpu::ChargeRatioElectrons, pmacc::pmacc_isAlias> >, boost::mp11::mp_list<picongpu::position<picongpu::position_pic, pmacc::pmacc_isAlias>, picongpu::momentum, picongpu::weighting> >; Filter = picongpu::particles::filter::All; uint32_t = unsigned int]’:
/lustre/rz/dbertini/hydra/pog/picongpu/include/picongpu/plugins/openPMD/openPMDWriter.x.cpp:791:1:   required from ‘void picongpu::openPMD::openPMDWriter::GetFields<picongpu::FieldTmpOperation<Solver, Species, Filter> >::operator()(picongpu::openPMD::ThreadParams*, uint32_t) [with Solver = picongpu::particles::particleToGrid::ComputeGridValuePerFrame<picongpu::particles::shapes::TSC, picongpu::particles::particleToGrid::derivedAttributes::EnergyDensity>; Species = picongpu::Particles<pmacc::meta::String<'e'>, boost::mp11::mp_list<picongpu::particlePusher<picongpu::particles::pusher::Boris, pmacc::pmacc_isAlias>, picongpu::shape<picongpu::particles::shapes::TSC, pmacc::pmacc_isAlias>, picongpu::interpolation<picongpu::FieldToParticleInterpolation<picongpu::particles::shapes::TSC, picongpu::AssignedTrilinearInterpolation>, pmacc::pmacc_isAlias>, picongpu::current<picongpu::currentSolver::Esirkepov<picongpu::particles::shapes::TSC, picongpu::currentSolver::strategy::CachedSupercells, 3>, pmacc::pmacc_isAlias>, picongpu::massRatio<picongpu::MassRatioElectrons, pmacc::pmacc_isAlias>, picongpu::chargeRatio<picongpu::ChargeRatioElectrons, pmacc::pmacc_isAlias> >, boost::mp11::mp_list<picongpu::position<picongpu::position_pic, pmacc::pmacc_isAlias>, picongpu::momentum, picongpu::weighting> >; Filter = picongpu::particles::filter::All; uint32_t = unsigned int]’
/lustre/rz/dbertini/hydra/pog/picongpu/include/picongpu/../pmacc/meta/ForEach.hpp:72:13:   required from ‘void pmacc::meta::ForEach<List, T_Functor, T_Accessor>::callEachFunctorWithArgs(boost::mp11::mp_list<U ...>, TArgs&& ...) const [with TFunctors = {picongpu::openPMD::openPMDWriter::GetFields<picongpu::FieldE>, picongpu::openPMD::openPMDWriter::GetFields<picongpu::FieldB>, picongpu::openPMD::openPMDWriter::GetFields<picongpu::FieldTmpOperation<picongpu::particles::particleToGrid::ComputeGridValuePerFrame<picongpu::particles::shapes::TSC, picongpu::particles::particleToGrid::derivedAttributes::ChargeDensity>, picongpu::Particles<pmacc::meta::String<'e'>, boost::mp11::mp_list<picongpu::particlePusher<picongpu::particles::pusher::Boris, pmacc::pmacc_isAlias>, picongpu::shape<picongpu::particles::shapes::TSC, pmacc::pmacc_isAlias>, picongpu::interpolation<picongpu::FieldToParticleInterpolation<picongpu::particles::shapes::TSC, picongpu::AssignedTrilinearInterpolation>, pmacc::pmacc_isAlias>, picongpu::current<picongpu::currentSolver::Esirkepov<picongpu::particles::shapes::TSC, picongpu::currentSolver::strategy::CachedSupercells, 3>, pmacc::pmacc_isAlias>, picongpu::massRatio<picongpu::MassRatioElectrons, pmacc::pmacc_isAlias>, picongpu::chargeRatio<picongpu::ChargeRatioElectrons, pmacc::pmacc_isAlias> >, boost::mp11::mp_list<picongpu::position<picongpu::position_pic, pmacc::pmacc_isAlias>, picongpu::momentum, picongpu::weighting> >, picongpu::particles::filter::All> >, picongpu::openPMD::openPMDWriter::GetFields<picongpu::FieldTmpOperation<picongpu::particles::particleToGrid::ComputeGridValuePerFrame<picongpu::particles::shapes::TSC, picongpu::particles::particleToGrid::derivedAttributes::EnergyDensity>, picongpu::Particles<pmacc::meta::String<'e'>, boost::mp11::mp_list<picongpu::particlePusher<picongpu::particles::pusher::Boris, pmacc::pmacc_isAlias>, picongpu::shape<picongpu::particles::shapes::TSC, pmacc::pmacc_isAlias>, picongpu::interpolation<picongpu::FieldToParticleInterpolation<picongpu::particles::shapes::TSC, picongpu::AssignedTrilinearInterpolation>, pmacc::pmacc_isAlias>, picongpu::current<picongpu::currentSolver::Esirkepov<picongpu::particles::shapes::TSC, picongpu::currentSolver::strategy::CachedSupercells, 3>, pmacc::pmacc_isAlias>, picongpu::massRatio<picongpu::MassRatioElectrons, pmacc::pmacc_isAlias>, picongpu::chargeRatio<picongpu::ChargeRatioElectrons, pmacc::pmacc_isAlias> >, boost::mp11::mp_list<picongpu::position<picongpu::position_pic, pmacc::pmacc_isAlias>, picongpu::momentum, picongpu::weighting> >, picongpu::particles::filter::All> >}; TArgs = {picongpu::openPMD::ThreadParams*&, const unsigned int&}; List = boost::mp11::mp_list<picongpu::FieldE, picongpu::FieldB, picongpu::FieldTmpOperation<picongpu::particles::particleToGrid::ComputeGridValuePerFrame<picongpu::particles::shapes::TSC, picongpu::particles::particleToGrid::derivedAttributes::ChargeDensity>, picongpu::Particles<pmacc::meta::String<'e'>, boost::mp11::mp_list<picongpu::particlePusher<picongpu::particles::pusher::Boris, pmacc::pmacc_isAlias>, picongpu::shape<picongpu::particles::shapes::TSC, pmacc::pmacc_isAlias>, picongpu::interpolation<picongpu::FieldToParticleInterpolation<picongpu::particles::shapes::TSC, picongpu::AssignedTrilinearInterpolation>, pmacc::pmacc_isAlias>, picongpu::current<picongpu::currentSolver::Esirkepov<picongpu::particles::shapes::TSC, picongpu::currentSolver::strategy::CachedSupercells, 3>, pmacc::pmacc_isAlias>, picongpu::massRatio<picongpu::MassRatioElectrons, pmacc::pmacc_isAlias>, picongpu::chargeRatio<picongpu::ChargeRatioElectrons, pmacc::pmacc_isAlias> >, boost::mp11::mp_list<picongpu::position<picongpu::position_pic, pmacc::pmacc_isAlias>, picongpu::momentum, picongpu::weighting> >, picongpu::particles::filter::All>, picongpu::FieldTmpOperation<picongpu::particles::particleToGrid::ComputeGridValuePerFrame<picongpu::particles::shapes::TSC, picongpu::particles::particleToGrid::derivedAttributes::EnergyDensity>, picongpu::Particles<pmacc::meta::String<'e'>, boost::mp11::mp_list<picongpu::particlePusher<picongpu::particles::pusher::Boris, pmacc::pmacc_isAlias>, picongpu::shape<picongpu::particles::shapes::TSC, pmacc::pmacc_isAlias>, picongpu::interpolation<picongpu::FieldToParticleInterpolation<picongpu::particles::shapes::TSC, picongpu::AssignedTrilinearInterpolation>, pmacc::pmacc_isAlias>, picongpu::current<picongpu::currentSolver::Esirkepov<picongpu::particles::shapes::TSC, picongpu::currentSolver::strategy::CachedSupercells, 3>, pmacc::pmacc_isAlias>, picongpu::massRatio<picongpu::MassRatioElectrons, pmacc::pmacc_isAlias>, picongpu::chargeRatio<picongpu::ChargeRatioElectrons, pmacc::pmacc_isAlias> >, boost::mp11::mp_list<picongpu::position<picongpu::position_pic, pmacc::pmacc_isAlias>, picongpu::momentum, picongpu::weighting> >, picongpu::particles::filter::All> >; T_Functor = picongpu::openPMD::openPMDWriter::GetFields<mpl_::arg<1> >; T_Accessor = pmacc::meta::accessors::Identity<>]’
/lustre/rz/dbertini/hydra/pog/picongpu/include/picongpu/../pmacc/meta/ForEach.hpp:64:24:   required from ‘void pmacc::meta::ForEach<List, T_Functor, T_Accessor>::operator()(T_Types&& ...) const [with T_Types = {picongpu::openPMD::ThreadParams*&, const unsigned int&}; List = boost::mp11::mp_list<picongpu::FieldE, picongpu::FieldB, picongpu::FieldTmpOperation<picongpu::particles::particleToGrid::ComputeGridValuePerFrame<picongpu::particles::shapes::TSC, picongpu::particles::particleToGrid::derivedAttributes::ChargeDensity>, picongpu::Particles<pmacc::meta::String<'e'>, boost::mp11::mp_list<picongpu::particlePusher<picongpu::particles::pusher::Boris, pmacc::pmacc_isAlias>, picongpu::shape<picongpu::particles::shapes::TSC, pmacc::pmacc_isAlias>, picongpu::interpolation<picongpu::FieldToParticleInterpolation<picongpu::particles::shapes::TSC, picongpu::AssignedTrilinearInterpolation>, pmacc::pmacc_isAlias>, picongpu::current<picongpu::currentSolver::Esirkepov<picongpu::particles::shapes::TSC, picongpu::currentSolver::strategy::CachedSupercells, 3>, pmacc::pmacc_isAlias>, picongpu::massRatio<picongpu::MassRatioElectrons, pmacc::pmacc_isAlias>, picongpu::chargeRatio<picongpu::ChargeRatioElectrons, pmacc::pmacc_isAlias> >, boost::mp11::mp_list<picongpu::position<picongpu::position_pic, pmacc::pmacc_isAlias>, picongpu::momentum, picongpu::weighting> >, picongpu::particles::filter::All>, picongpu::FieldTmpOperation<picongpu::particles::particleToGrid::ComputeGridValuePerFrame<picongpu::particles::shapes::TSC, picongpu::particles::particleToGrid::derivedAttributes::EnergyDensity>, picongpu::Particles<pmacc::meta::String<'e'>, boost::mp11::mp_list<picongpu::particlePusher<picongpu::particles::pusher::Boris, pmacc::pmacc_isAlias>, picongpu::shape<picongpu::particles::shapes::TSC, pmacc::pmacc_isAlias>, picongpu::interpolation<picongpu::FieldToParticleInterpolation<picongpu::particles::shapes::TSC, picongpu::AssignedTrilinearInterpolation>, pmacc::pmacc_isAlias>, picongpu::current<picongpu::currentSolver::Esirkepov<picongpu::particles::shapes::TSC, picongpu::currentSolver::strategy::CachedSupercells, 3>, pmacc::pmacc_isAlias>, picongpu::massRatio<picongpu::MassRatioElectrons, pmacc::pmacc_isAlias>, picongpu::chargeRatio<picongpu::ChargeRatioElectrons, pmacc::pmacc_isAlias> >, boost::mp11::mp_list<picongpu::position<picongpu::position_pic, pmacc::pmacc_isAlias>, picongpu::momentum, picongpu::weighting> >, picongpu::particles::filter::All> >; T_Functor = picongpu::openPMD::openPMDWriter::GetFields<mpl_::arg<1> >; T_Accessor = pmacc::meta::accessors::Identity<>]’
/lustre/rz/dbertini/hydra/pog/picongpu/include/picongpu/plugins/openPMD/openPMDWriter.x.cpp:1860:17:   required from here
/lustre/rz/dbertini/hydra/pog/picongpu/include/picongpu/plugins/openPMD/openPMDWriter.x.cpp:870:140: error: dependent-name ‘pmacc::traits::GetComponentsType<pmacc::math::Vector<float, 1, pmacc::math::ArrayStorage<float, 1> >, is_fundamental_v<pmacc::math::Vector<float, 1, pmacc::math::ArrayStorage<float, 1> > > >::type’ is parsed as a non-type, but instantiation yields a type
  870 |                     openPMDWriter::template writeField<ComponentType>(
      |                                                                                                                                            ^   
/lustre/rz/dbertini/hydra/pog/picongpu/include/picongpu/plugins/openPMD/openPMDWriter.x.cpp:870:140: note: say ‘typename pmacc::traits::GetComponentsType<pmacc::math::Vector<float, 1, pmacc::math::ArrayStorage<float, 1> >, is_fundamental_v<pmacc::math::Vector<float, 1, pmacc::math::ArrayStorage<float, 1> > > >::type’ if a type is meant
/lustre/rz/dbertini/hydra/pog/picongpu/include/picongpu/plugins/openPMD/openPMDWriter.x.cpp:870:146: error: no matching function for call to ‘picongpu::openPMD::openPMDWriter::<expression error>(picongpu::openPMD::ThreadParams*&, const uint32_t&, const uint32_t&, std::string, std::__shared_ptr_access<picongpu::FieldTmp, __gnu_cxx::_S_atomic, false, false>::element_type&, std::vector<double>, std::vector<double>, std::remove_reference<std::vector<std::vector<float> >&>::type, const float_X&, const bool&, bool)’
  870 |                     openPMDWriter::template writeField<ComponentType>(
      |                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                                                           ^                                                                                                                                                                                     
/lustre/rz/dbertini/hydra/pog/picongpu/include/picongpu/plugins/openPMD/openPMDWriter.x.cpp:1532:1: note: candidate: ‘template<class ComponentType, class FieldBuffer> static void picongpu::openPMD::openPMDWriter::writeField(picongpu::openPMD::ThreadParams*, uint32_t, uint32_t, std::string, FieldBuffer&, std::vector<double>, std::vector<double>, std::vector<std::vector<float> >, picongpu::float_X, bool, bool)’
 1532 |             static void writeField(
      | ^         
/lustre/rz/dbertini/hydra/pog/picongpu/include/picongpu/plugins/openPMD/openPMDWriter.x.cpp:1532:1: note:   template argument deduction/substitution failed:
gmake[2]: *** [CMakeFiles/picongpu.dir/build.make:410: CMakeFiles/picongpu.dir/plugins/openPMD/openPMDWriter.x.cpp.o] Error 1
gmake[2]: *** Waiting for unfinished jobs....
gmake[1]: *** [CMakeFiles/Makefile2:306: CMakeFiles/picongpu.dir/all] Error 2
gmake: *** [Makefile:136: all] Error 2
```
any idea what could go wrong here, i use here the last version of  `openPMD-api 0.16.1`

Note on what makes me think that this is a compiler bug:

```
/lustre/rz/dbertini/hydra/pog/picongpu/include/picongpu/plugins/openPMD/openPMDWriter.x.cpp:870:140: error: dependent-name ‘pmacc::traits::GetComponentsType<pmacc::math::Vector<float, 1, pmacc::math::ArrayStorage<float, 1> >, is_fundamental_v<pmacc::math::Vector<float, 1, pmacc::math::ArrayStorage<float, 1> > > >::type’ is parsed as a non-type, but instantiation yields a type

  870 |                     openPMDWriter::template writeField<ComponentType>(
```
The type definition that the compiler is complaining about here is not in the code in that form, but it is the expanded form of `ComponentType`. This seems like the compiler is inlining the type definition into that line, and introduces a syntax error (missing typename) while doing this.

--- a/include/pmacc/traits/GetComponentsType.hpp
+++ b/include/pmacc/traits/GetComponentsType.hpp
@@ -35,8 +35,11 @@ namespace pmacc
          *            If T_Type is fundamental c++ type, the identity is returned
          *
          * Attention: do not defines this trait for structs with different attributes inside
+         *
+         * Intentionally not using std::is_fundamental_v<T_Type> due to a compiler bug in cuda-13/gcc-11 combination,
+         * ref. https://github.com/ComputationalRadiationPhysics/picongpu/issues/5554
          */
-        template<typename T_Type, bool T_IsFundamental = std::is_fundamental_v<T_Type>>
+        template<typename T_Type, bool T_IsFundamental = std::is_fundamental<T_Type>::value>
         struct GetComponentsType;
 
         template<typename T_Type>

